File: 2022年9月/20 349. 两个数组的交集/intersection.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 * 执行用时：68 ms, 在所有 JavaScript 提交中击败了36.36%的用户
 * 内存消耗：42.9 MB, 在所有 JavaScript 提交中击败了16.60%的用户
 */
var intersection = function (nums1, nums2) {
    const set = new Set();
    for (let num of nums1) {
        set.add(num);
    }
    const res = new Set();
    for (let num of nums2) {
        if (set.has(num)) {
            res.add(num);
        }
    }
    return Array.from(res);
};

File: 2022年9月/20 349. 两个数组的交集/intersection.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        执行用时：20 ms, 在所有 Python 提交中击败了63.39%的用户
        内存消耗：13.3 MB, 在所有 Python 提交中击败了21.95%的用户
        """
        return list(set(nums1) & set(nums2))